Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1429,15 +1429,24 @@ function Set-HcxScaledCpuAndMemorySetting {
 
 <#
     .Synopsis
-     This will create a folder on every datastore (/vmfs/volumes/datastore/tools-repo) and set the ESXi hosts to use that folder as the tools-repo. The customer is responsible for putting the VMware Tools files into the folder.
+     This will create a folder on every datastore (/vmfs/volumes/datastore/tools-repo) and set the ESXi hosts to use that folder as the tools-repo.
+     The customer is responsible for putting the VMware Tools zip file in a downloadable location.
 
      .EXAMPLE
-     Once the function is imported, you simply need to run Set-ToolsRepo without any parameters.
+     Once the function is imported, you simply need to run Set-ToolsRepo -url <url to tools zip file>
 #>
 
 function Set-ToolsRepo
 {
-    param()
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]
+        $url
+    )
+
+    # Make sure no old files exist
+    Remove-Item -Path "./newtools" -Recurse -ErrorAction SilentlyContinue
```

Comment:
Each execution gets a new temp folder, there's no need to cleanup.

Reply:
Done